### PR TITLE
Don't show no patients found error view when search query is empty

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.shortcut.com/simpledotorg/story/8601/don-t-show-no-patients-found-error-view-when-search-query-is-empty

### CODE REVIEW

**Comment by msasikanth** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4025** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/4025" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/4025?utm_source=stack-comment).

File: app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchEvent.kt [hunk 1]
```diff
@@ -21,3 +21,5 @@ data class OverduePatientClicked(val patientUuid: UUID) : OverdueSearchEvent()
 data class CallPatientClicked(val patientUuid: UUID) : OverdueSearchEvent()
 
 data class OverdueSearchHistoryClicked(val searchQuery: String) : OverdueSearchEvent()
+
+data class OverdueSearchLoadStateChanged(val overdueSearchProgressState: OverdueSearchProgressState) : OverdueSearchEvent()
```
Comment (janhavisinghh) [inline app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchEvent.kt:25]: If we are naming the enum as `OverdueSearchProgressState`  then let's name the event as `OverdueSearchProgressStateChanged` instead?

File: app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchModel.kt [hunk 2]
```diff
@@ -31,4 +33,8 @@ data class OverdueSearchModel(
   fun overdueSearchQueryChanged(searchQuery: String): OverdueSearchModel {
     return copy(searchQuery = searchQuery)
   }
+
+  fun loadStateChanged(overdueSearchProgressState: OverdueSearchProgressState): OverdueSearchModel {
```
Comment (janhavisinghh) [inline app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchModel.kt:37]: Same comment as above. Let's rename this to `progressStateChanged`

**Review by janhavisinghh — CHANGES_REQUESTED** [review]
See inline comments

File: app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchEvent.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchEvent.kt:25]: So, this event is triggered when the paging adapter load state is changed. So, we are following that naming convention. Similar to what we did in instant search screen.

File: app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchModel.kt (hunk 2 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchModel.kt:37]: Will update this

File: app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchEvent.kt (hunk 1 above)
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchEvent.kt:25]: umm okay

**Review by janhavisinghh — APPROVED** [review]

**Pull request merged**